jupyterhub/handlers/base.py

```diff
@@ -1448,24 +1449,18 @@ def non_get(self, user_name, user_path):
     delete = non_get
 
     @web.authenticated
+    @needs_scope("access:servers")
     async def get(self, user_name, user_path):
         if not user_path:
             user_path = '/'
         current_user = self.current_user
-
-        if (
-            current_user
-            and current_user.name != user_name
-            and current_user.admin
-            and self.settings.get('admin_access', False)
-        ):
-            # allow admins to spawn on behalf of users
+        if user_name != current_user.name:
             user = self.find_user(user_name)
             if user is None:
                 # no such user
                 raise web.HTTPError(404, "No such user %s" % user_name)
             self.log.info(
-                "Admin %s requesting spawn on behalf of %s",
+                f"User {current_user.name} requesting spawn on behalf of {user.name}",
                 current_user.name,
                 user.name,
```

Comment:
You still pass log the objects part of the f-string!

Reply:
Another good catch; fixed!